File: packages/capp-sdk-ts/src/modules/compliance.ts
import { CAPPClient } from '../client';
import { handleApiError } from '../utils';

export class ComplianceModule {
    constructor(private client: CAPPClient) { }

    public async getCsvReport(organizationId: string): Promise<string> {
        /**
         * Downloads the full transaction history for the organization as CSV.
         * Returns the CSV file content as a string.
         */
        const res = await this.client.fetch(`/compliance/reports/csv?organizationId=${organizationId}`);
        await handleApiError(res);
        return res.text();
    }
}

File: packages/capp-sdk/capp/modules/compliance.py
import httpx
from .._utils import handle_api_error

class ComplianceModule:
    def __init__(self, client: httpx.AsyncClient):
        self._client = client

    async def get_csv_report(self, organization_id: str) -> str:
        """
        Downloads the full transaction history for the organization as CSV.
        Returns the CSV file content as a string.
        """
        res = await self._client.get(f"/compliance/reports/csv?organization_id={organization_id}")
        handle_api_error(res)
        return res.text
